sklearn/metrics/classification.py

```diff
@@ -91,6 +91,10 @@ def _check_targets(y_true, y_pred):
     if y_type in ["binary", "multiclass"]:
         y_true = column_or_1d(y_true)
         y_pred = column_or_1d(y_pred)
+        if y_type == "binary":
```

Comment:
This is the less invasive fix I could find. This is not super efficient because `np.unique` has been already called on `y_true` and `y_pred` to figure out their type in `type_of_target`.